components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -86,8 +86,28 @@ abstract class AbstractFetchDownloadService : Service() {
         @GuardedBy("context") var status: DownloadJobStatus,
         var foregroundServiceId: Int = 0,
         var downloadDeleted: Boolean = false,
-        var notifiedStopped: Boolean = false
-    )
+        var notifiedStopped: Boolean = false,
+        var lastNotificationUpdate: Long = 0L
+    ) {
+        internal fun canUpdateNotification(): Boolean {
+            return isUnderNotificationUpdateLimit() && !notifiedStopped
+        }
+
+        /**
+         * Android imposes on limit of how often we can send updates for a notification.
```

Comment:
nit: imposes a limit on of how often...:)

Reply:
👍